hnn_core/network.py

```diff
@@ -245,53 +246,54 @@ def pick_connection(net, src_gids=None, target_gids=None,
     valid_receptor = ['ampa', 'nmda', 'gabaa', 'gabab']
 
     # Convert receptor and loc to list
-    loc = _string_input_to_list(loc, valid_loc, 'loc')
-    receptor = _string_input_to_list(receptor, valid_receptor, 'receptor')
+    loc_list = _string_input_to_list(loc, valid_loc, 'loc')
+    receptor_list = _string_input_to_list(receptor, valid_receptor, 'receptor')
```

Comment:
I know the `src_dict` already specifies the object type in the variable name but in general, I try to avoid that. It should be apparent from the [object's methods what it is](https://en.wikipedia.org/wiki/Duck_typing)

Reply:
I'm not sure I agree on this point as it doesn't help readability. You're not always in an interactive environment to check on a function's methods and so a reviewer needs to expend cognitive load to decipher type. And that ability is wholely dependent on if the writer uses a class-specific method at some point. It also doesn't help that because python is dynamically typed, type can change... Just like in this example here where a string is now reassigned as a list. 

I believe that the Duck typing concept is not about variable naming but about function design with regard to its behavior at compile time. The function doesn't need to explicitly define or restrict the object data types it accepts. As long as the object contains methods the function uses it should work, and if it doesn't contain it, it should fail gracefully.

Reply:
It's fine to use the type in the object name sparingly for variables whose type changes and there is a need to disambiguate. I generally find it overkill for other use cases as I feel Python code needs to read like English. 

But let's keep it here as it seems to be the former case.